spyder/plugins/editor/widgets/tests/test_warnings.py

```diff
@@ -175,3 +177,71 @@ def test_update_warnings_after_delete_line(qtbot, lsp_codeeditor):
                 ['E261 at least two spaces before inline comment', 2],
                 ["undefined name 's'", 4]]
     assert editor.get_current_warnings() == expected
+
+
+@pytest.mark.slow
+@pytest.mark.second
+def test_update_warnings_after_closequotes(qtbot, lsp_codeeditor):
+    """
+    Test that code errors are correctly updated after activating closequotes
+    in the Editor.
+
+    Regression test for #9323.
+    """
+    editor, _ = lsp_codeeditor
+    editor.textCursor().insertText("print('test)\n")
+
+    expected = [['EOL while scanning string literal', 1]]
+
+    # Notify changes.
+    with qtbot.waitSignal(editor.lsp_response_signal, timeout=30000):
+        editor.document_did_change()
+
+    assert editor.get_current_warnings() == expected
+
+    # Add a single quote to fix the error
+    editor.move_cursor(-2)
+    qtbot.keyPress(editor, Qt.Key_Apostrophe)
+    assert editor.toPlainText() == "print('test')\n"
+
+    # Wait for the lsp_response_signal.
+    with qtbot.waitSignal(editor.lsp_response_signal, timeout=30000):
+        sleep(1)
```

Comment:
Why is this necessary? I think instead of this you need to run `editor.document_did_change()`.

Reply:
I'm not entirely  sure honestly but in my mind I'm  waiting for the server to be done processing.

> I think instead of this you need to run editor.document_did_change()

The only change I made to the the spyder code other than tests was adding editor.document_did_change() to close quotes and brackets. I don't  know how editor.document_did_change() works but if I just call editor.document_did_change() here wouldn't that defeat the purpose of the test?

Reply:
I agree with @bcolsen. I think if we are using `editor.document_did_change()` on the tests, we are probably doing something wrong as the editor should be doing this "on its own". 

Now the way to use the context blockers should be:

```python
with qtbot.waitSignal(editor.lsp_response_signal, timeout=30000):
    # do something (add a quote.... remove a quote) no need to add waits
```

I suggest you put the code that triggers something inside the `with` instead of document change or a wait. The code inside the with should be triggering some signal.

Reply:
I tried that first, but I couldn't get it to fail and I needed to add the sleep in side as well to make it pass.

Reply:
@bcolsen, ok, you are the one who fought with the test, so I defer to you on this one. Please add a comment above this line then, explaining why it's needed.

Also, why didn't you use `qtbot.wait` instead?

Reply:
I didn't try qtbot.wait. That should work.

Reply:
Yeah, I think it's better because it's non-blocking (while `sleep` is).

Reply:
Yep! correct. 

@bcolsen I will check later what might be the problem that requires manually calling `document_did_change` Thanks for the work here!